### PR TITLE
ADR 13: Repository Layer updates / edits

### PR DESCRIPTION
**Pull request by rsanheim**
[ch3100](https://app.clubhouse.io/simpledotorg/story/3100/repository-adr)

The initial draft of this ADR was merged w/ previous repository work (see #2059), so that PR may give more context on the usage. The best way to view the file is probably with the [full rendered view **here**](https://github.com/simpledotorg/simple-server/blob/72b73ddbd6cf52ac0b2d3628c34cc478755f0893/doc/arch/013-repository-layer-for-reporting-data.md).

Going to do an editing pass here as well as handle questions / feedback.

### CODE REVIEW

**Review by kpethtel — APPROVED** [review]
👍

**Review by prabhanshuguptagit — APPROVED** [review]
:shipit:. I'll let someone unfamiliar with this (since I've been pretty much knee deep in this code) comment as well.

File: doc/arch/013-repository-layer-for-reporting-data.md [hunk 1]
```diff
@@ -16,7 +16,10 @@ Our current reports retrieve their data in a variety of ways:
 We will introduce a [Repository](https://martinfowler.com/eaaCatalog/repository.html) layer that will be responsible for returning reporting data. It will provide a consistent
 interface where callers provide one to many Regions, and one to many Periods, and then can retreive whatever values they want from the Repository for the intersection of those Regions and Periods.
 
-The Repository will be responsible for grabbing the values from whatever queries, calculations, or data sources are necessary, and will also handle any rqeuired caching to make sure things are performant.
+The Repository will be responsible for grabbing the values from whatever queries, calculations, or data sources are necessary, and will also handle any required caching to make sure things are performant.
+
+The caching and query implementation will be entirely handled by the Repository (plus any related objects helping), and will be transparent to callers. In other words, callers should be able to ask for
```
[the remark this answers is not in the document]
Comment (rsanheim) [reply doc/arch/013-repository-layer-for-reporting-data.md:21]: yup good catch

File: doc/arch/013-repository-layer-for-reporting-data.md [hunk 2]
```diff
@@ -16,7 +16,10 @@ Our current reports retrieve their data in a variety of ways:
 We will introduce a [Repository](https://martinfowler.com/eaaCatalog/repository.html) layer that will be responsible for returning reporting data. It will provide a consistent
 interface where callers provide one to many Regions, and one to many Periods, and then can retreive whatever values they want from the Repository for the intersection of those Regions and Periods.
 
-The Repository will be responsible for grabbing the values from whatever queries, calculations, or data sources are necessary, and will also handle any rqeuired caching to make sure things are performant.
+The Repository will be responsible for grabbing the values from whatever queries, calculations, or data sources are necessary, and will also handle any required caching to make sure things are performant.
```
[the remark this answers is not in the document]
Comment (rsanheim) [reply doc/arch/013-repository-layer-for-reporting-data.md]: added 👍

**Pull request merged**